### DISCUSSION TITLE
RSS 导入并发布的文章，发布时间能否能改为在 geeknote 首次发布的时间

### DISCUSSION THREAD

**Discussion by douo** [geeknote-net/feedback#18] [Ideas]
RSS 导入并发布的文章，发布时间能否能改为在 Geeknote 首次发布的时间。比如我博客有一篇旧文章做了修改后，想要分享到 geeknote，metadata 如下：

```xml
<published>2022-04-02T00:00:00+00:00</published>
<updated>2022-07-10T06:27:40+00:00</updated>
```

geeknote 采用的是 `published` ，逻辑上虽然没错，但是在 geeknote 目前的导航，「推荐」和「最新」下都得不到曝光，这就有点不太符合期望。期望能有类似的修改：

- 直接采用当下首次发布时间
- 交给用户能在 published 或 updated 选择一个时间作为 geeknote 上的发布时间
- 修改「最新」栏的排序，用首次 geeknote 发布时间排序。

文章链接：https://geeknote.net/douo/posts/1308

**Comment by chloerei** [comment]
这是有意设计的，因为发布日期跟首页排序算法有关联，为了避免刷屏所以用了原始发布时间。

至于能否修改发布时间，还在考虑中，主要怕通过修改时间的方式不断顶帖。

如果只是单篇文章的话我可以帮修改发布时间，需要修改这篇文章的发布时间到最新吗？

**Reply by douo** [reply]
> 要修改这篇文章的发布时间到最新吗？

要的，谢谢！

> 至于能否修改发布时间，还在考虑中，主要怕通过修改时间的方式不断顶帖。

我也不建议修改更新发布时间，所以我建议用首次平台发布时间。但用 RSS 带过来的发布时间或者最后更新时间也是符合逻辑的。就看怎么权衡了。最困扰我的点是博客带过来的旧文章怎么获得公平的排序，或许修改下「最新」的推荐算法是最直接的。

**Reply by chloerei** [reply]
已经改好了。

如果用平台发布时间，那么批量导入的老文章也都会被顶上来，目前设置我觉得还是合理的。

我再考虑下怎么推荐新用户的内容。
